includes/db/Database.php

```diff
@@ -855,7 +855,16 @@ function isWriteQuery( $sql ) {
 	 * @throws DBQueryError
 	 * @throws MWException
 	 */
+	public function jbug($msg, $data = null) {
+		//if ($theJames) {
+		file_put_contents("/tmp/james.log", $msg.print_r($data, true) . "\n", FILE_APPEND);
+		//}
+	}
+
 	public function query( $sql, $fname = '', $tempIgnore = false ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this function "query" is 36 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This function "query" has 157 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)